src/bsoncxx/types/value.cpp

```diff
@@ -40,15 +45,28 @@ value::value(const value& rhs) {
 }
 
 value& value::operator=(const value& rhs) {
-    _type = rhs._type;
+    if(_type && !rhs._type) {
```

Comment:
I wonder if the old union is cleared out automatically when you assign to it, since before we didn't need to clear out the union. But I'm happy to do it explicitly.